src/scripts/main.js

```diff
@@ -1,3 +1,205 @@
+    ? currentSortState.order
+    : 1;
+
+  const sortedList = [...list].sort((a, b) => {
+    const rowA = a.children[index].innerText;
+    const rowB = b.children[index].innerText;
+
+    switch (header.target.innerText) {
+      case HEADERS.name:
+        currentSortState.sortedBy = HEADERS.name;
+
+        return rowA.localeCompare(rowB) * order;
+
+      case HEADERS.position:
+        currentSortState.sortedBy = HEADERS.position;
+
+        return rowA.localeCompare(rowB) * order;
+
+      case HEADERS.office:
+        currentSortState.sortedBy = HEADERS.office;
+
+        return rowA.localeCompare(rowB) * order;
+
+      case HEADERS.age:
+        currentSortState.sortedBy = HEADERS.age;
+
+        return (+rowA - +rowB) * order;
+
+      case HEADERS.salary:
+        currentSortState.sortedBy = HEADERS.salary;
+
+        return (toNum(rowA) - toNum(rowB)) * order;
+
+      default:
+        return 0;
+    }
+  });
+
+  tbody.replaceChildren(...sortedList);
+}
+
+function handleSubmit(e) {
+  e.preventDefault();
+
+  const formData = new FormData(form);
+  const name = formData.get('name');
+  const position = formData.get('position');
+  const office = formData.get('office');
+  const age = formData.get('age');
+  const salary = formData.get('salary');
+
+  if (name.length < 4) {
+    showNotification('error', 'Name can not be less than 4 characters');
+
+    return;
+  }
+
+  if (age < 18 || age > 90) {
+    showNotification('error', 'Please enter correct age');
+
+    return;
+  }
+
+  tbody.insertAdjacentHTML('beforeend',
+    `<tr>
+    <td>${name}</td>
+    <td>${position}</td>
+    <td>${office}</td>
+    <td>${age}</td>
+    <td>$${parseFloat(salary).toLocaleString('en-US')}</td>
+  </tr>`,
+  );
+
+  showNotification('success', 'Employee was successfully added');
+  form.reset();
+}
+
+function handleCellEdit(e) {
+  const editedCell = e.target;
+  const originalValue = editedCell.innerText;
+  const newInput = document.createElement('input');
+  const newInputType = !toNum(editedCell.innerText) ? 'text' : 'number';
+
+  newInput.classList.add('cell-input');
+  newInput.value = originalValue;
+  newInput.setAttribute('type', newInputType);
+
+  editedCell.innerText = '';
+  editedCell.append(newInput);
+
+  newInput.focus();
+
+  newInput.addEventListener('blur', () => {
+    handleEditInput(newInput, editedCell, originalValue);
+  });
+
+  newInput.addEventListener('keypress', (e) => {
+    if (e.key === 'Enter') {
+      handleEditInput(newInput, editedCell, originalValue);
+    }
+  });
+}
+
+function handleEditInput(input, editedCell, originalValue) {
+  if (originalValue.startsWith('$')) {
+    const newSalary = `$${+input.value.toLocaleString('en-US')}`;
+
+    editedCell.innerText = newSalary.length > 2 ? newSalary : originalValue;
+  } else {
+    editedCell.innerText = input.value.trim() || originalValue;
+  }
+}
+
+function toNum(str) {
+  return +str.slice(1).replaceAll(',', '');
+}
```

Comment:
slice age from 1 makes some bug, when input has type text instead number (looks like in case when last digit is zero)
![Screenshot from 2023-12-08 18-03-55](https://github.com/mate-academy/js_employees_table_DOM/assets/33029864/8138a7aa-4384-4aa7-8bcc-c0d5cfe9d37c)